### DISCUSSION TITLE
Testing in a docker container

### DISCUSSION THREAD

**Discussion by adamgoryachev** [LINBIT/linstor-gui#1] [Q&A]
Was looking to test this inside a docker container, and am not familiar with nodejs apps, but here is my attempt and where I got to. Appreciate any suggestions/advice.
Using the docker image node:22, and starting with a shell.
# cd /root
# git clone https://github.com/LINBIT/linstor-gui.git
# cd linstor-gui/
# cat .env
HOST=127.0.0.1
PORT=8080
VERSION=DEV
LINSTOR_API_HOST=http://192.168.x.x:3370
# npm install --global cross-env
# npm run start:dev

> start:dev
> cross-env NODE_ENV=development webpack serve --config webpack.dev.js

node:events:498
      throw er; // Unhandled 'error' event
      ^

Error: spawn webpack ENOENT
    at ChildProcess._handle.onexit (node:internal/child_process:286:19)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21)
Emitted 'error' event on ChildProcess instance at:
    at ChildProcess._handle.onexit (node:internal/child_process:292:12)
    at onErrorNT (node:internal/child_process:484:16)
    at process.processTicksAndRejections (node:internal/process/task_queues:90:21) {
  errno: -2,
  code: 'ENOENT',
  syscall: 'spawn webpack',
  path: 'webpack',
  spawnargs: [ 'serve', '--config', 'webpack.dev.js' ]
}

Node.js v22.8.0

Happy to hear any ideas/suggestions on how to get this working?

**Comment by Liang0028** [comment]
Currently, it is recommended to run the UI in the LINSTOR controller using the installation package method. The original purpose of the development mode was to facilitate local development and similar tasks.

Although this is not a best practice and is intended for development, I have still conducted an experiment to run the development mode in Docker:

Here are the Dockerfile and commands:

```
# Use Node.js v22.8.0 as the base image
FROM node:22.8.0

# Set the working directory inside the container
WORKDIR /usr/src/app

# Install git in case it's not included in the base image
RUN apt-get update && apt-get install -y git

# Clone the repository
RUN git clone https://github.com/LINBIT/linstor-gui.git .

# Copy the environment variable configuration into the container
RUN echo "HOST=0.0.0.0" > .env && \
    echo "PORT=8080" >> .env && \
    echo "VERSION=DEV" >> .env && \
    echo "LINSTOR_API_HOST=http://192.168.123.105:3370" >> .env

# Install dependencies
RUN npm install

# Expose port 8080
EXPOSE 8080

# Set the default command to start the app in development mode
CMD ["npm", "run", "start:dev"]
```

1. Build the Docker image: `docker build -t linstor-gui .`
2. Run the container, binding port 8080: `docker run -p 8080:8080 linstor-gui`

I hope this information proves helpful to you. However, I would like to gently remind you once again that this is **not** the recommended approach.

**Comment by Liang0028** [comment]
The error message seems to indicate that the `npm install` step was not executed, or that npm couldn’t find `webpack` when running the command.
